### PR TITLE
Roll antags before job

### PR DESCRIPTION
**Pull request by Aviu00**
## About the PR

Now all antags except initial infected (there must be a delay for them so no zombies roundstart) are not delayed and are rolled before jobs. So you can pick a sec officer and you can still roll antag (your job will be different though).
Made that all intern roles can roll antag because why not.

Hopefully nothing breaks.

## Why / Balance

People asked for it.

## Media

https://github.com/user-attachments/assets/a910deb7-628a-4be3-af1b-19d481bf2ba6

## Requirements

- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

**Changelog**

:cl: Aviu
- add: Now antags are rolled roundstart and before jobs except for initial infected.
- add: Intern roles can be antags except for cadet obviously.

### CODE REVIEW

**Comment by whateverusername0** [comment]
WOAH

File: Resources/Prototypes/Roles/Jobs/Engineering/technical_assistant.yml [hunk 1]
```diff
@@ -13,7 +13,6 @@
   startingGear: TechnicalAssistantGear
   icon: "JobIconTechnicalAssistant"
   supervisors: job-supervisors-engineering
-  canBeAntag: false
```
Comment (whateverusername0) [inline Resources/Prototypes/Roles/Jobs/Engineering/technical_assistant.yml:16]: thank you

**Review by whateverusername0 — APPROVED** [review]

**Comment by Piras314** [comment]
```Robust.Shared.Utility.DebugAssertException : Rule Entity { Owner = 171253, Comp = Content.Server.Antag.Components.AntagSelectionComponent } had no spawner for pre-spawn rule added mid-round!```

**Comment by BombasterDS** [comment]
I can see problem with delay removal that players that won't press Ready button won't be able to become antags even if they joined first seconds after round start. Also, because you roll all antags PrePlayerSpawn late join antags won't spawn because round started already and they don't have spawners.

Imo, antagSelection should have they own delays

**Review by Piras314 — APPROVED** [review]

**Comment by Aviu00** [comment]
> I can see problem with delay removal that players that won't press Ready button won't be able to become antags even if they joined first seconds after round start.

So, just press ready button. Even if you don't press it you still can get latejoin antag.

>Also, because you roll all antags PrePlayerSpawn late join antags won't spawn because round started already and they don't have spawners.

Yeah, I fixed that.

> Imo, antagSelection should have they own delays to give antags to pending players.

That would probably require more messing with wizden code and I don't think it's that necessary.

**Comment by TurboTrackerss14** [comment]
I really don't think allowing interns to be antag is a bad change. 
They are fundamentally "learning" roles, to get your accustomed to a department. 
Having a reasonable chunk of your "learning" playtime be "oops actually I'm traitor time to murder" isn't really in line with a learner role. 

(However I would be fine with all interns becoming an antag selectable role if we had global playtime requirements for antags (like 10 hours before you can roll at all but I forgor if thats real at all)

**Comment by Piras314** [comment]
I don't think interns should be able to be antags honestly either.

**Comment by Aviu00** [comment]
> can you make it work with dynamic? @Aviu00

Maybe.

**Comment by Aviu00** [comment]
> I don't think interns should be able to be antags honestly either.

The point is that learning to play antags is an important part of learning to play ss14. If interns can't roll antags, they are missing an important aspect of the game. I think it's fine if they spend a few rounds as an antag. Also it would be weird if a person wants to play technical assistant but they roll antag before job and are forced to play assistant instead.

**Pull request merged**